src/apis/keycloak-api-service.ts

```diff
@@ -0,0 +1,90 @@
+import type { User } from "./schemas/keycloak/user";
+
+export interface KeycloakApiService {
+    getUsers: () => Promise<User[]>,
+    findUser: (id: string) => Promise<User[]>
+}
+
+/**
+ * Creates KeycloakApiService
+ */
+export function CreateKeycloakApiService(): KeycloakApiService {
+    const baseUrl: string = process.env.KEYCLOAK_BASE_URL
+    const realm: string = process.env.KEYCLOAK_REALM
+
+    return {
+        /**
+         * Gets all users from keycloak
+         * 
+         * @returns List of users
+         */
+        async getUsers(): Promise<User[]> {
+            const response = await fetch(`${baseUrl}/admin/realms/${realm}/users`, {
+                method: 'GET',
+                headers: {
+                    'Authorization': `Bearer ${await getAccessToken()}`
+                }
+            })
+
+            if (!response.ok) {
+                console.log(`Failed to fetch users: ${response.status} - ${response.statusText}`);
+            }
+
+            return response.json();
+        },
+
+        /**
+         * Find user from keycloak
+         * 
+         * @returns user by Id
+         */
+        async findUser(id: string): Promise<User[]> {
+            const response = await fetch(`${baseUrl}/admin/realms/${realm}/users/${id}`, {
+                method: 'GET',
+                headers: {
+                    'Authorization': `Bearer ${await getAccessToken()}`
+                } 
+            })
+
+            if (!response.ok) {
+                throw new Error(`Failed to find user with id: ${id}`);
```

Comment:
Be consistent across functions, either console.error or throw new Error but have the same behavior